drivers/rtc/rtc_emul.c

```diff
@@ -277,42 +265,41 @@ static int rtc_emul_set_time(const struct device *dev, const struct rtc_time *ti
 		return -EINVAL;
 	}
 
-	k_mutex_lock(&data->lock, K_FOREVER);
-
-	data->datetime = (*timeptr);
-	data->datetime.tm_isdst = -1;
-	data->datetime.tm_nsec = 0;
+	K_SPINLOCK_GUARDED_SECTION(&data->lock)
+	{
+		data->datetime = (*timeptr);
+		data->datetime.tm_isdst = -1;
+		data->datetime.tm_nsec = 0;
 
-	data->datetime_set = true;
-
-	k_mutex_unlock(&data->lock);
+		data->datetime_set = true;
+	}
 
 	return 0;
 }
 
 static int rtc_emul_get_time(const struct device *dev, struct rtc_time *timeptr)
 {
 	struct rtc_emul_data *data = (struct rtc_emul_data *)dev->data;
+	int ret = 0;
 
 	/* Validate arguments */
 	if (timeptr == NULL) {
 		return -EINVAL;
 	}
 
-	k_mutex_lock(&data->lock, K_FOREVER);
+	K_SPINLOCK_GUARDED_SECTION(&data->lock)
```

Comment:
Again - great pattern. Let's fork it out to a separate tree-wide PR and do it the old-fashioned way here to narrow the focus a bit.

Reply:
Ignored for now as not being part of this PR - see discussion below.